### ISSUE TITLE
access connection data from ProxyRequest Object

### ISSUE THREAD

**Issue by pauloFernandes** [Leanty/tree-gateway#159]
I was trying to write a request interceptor that would get client's IP via `req.connection.remoteAddress` and store it for usage from my services.

When I tried to use the `req.connection`, I found that the req object accessible in the interceptors doesn't have this data. Is there another way to get connection data in interceptors, or is it possible to add this data in the ProxyRequest object?

I searched for it and looks like this object is mounted [here](https://github.com/Leanty/tree-gateway/blob/master/src/pipeline/proxy/interceptor.ts#L77). Maybe, the connection data could be sent to interceptors via
```js
const proxyReq = (req as any).proxyReq || {
    body: req.body,
    headers: Object.assign({}, req.headers),
    method: req.method,
    url: req.url,
    user: req.user,
    connection: req.connection // Adding this field connection data from express req would be accessible in middleware req
};
```

Regards

**Comment by pauloFernandes** [comment]
Is this modification already published in npm? It still in [3.1.4](https://www.npmjs.com/package/tree-gateway) there

**Comment by thiagobustamante** [comment]
It is already published now.
